include/modem/nrf_modem_lib.rst

```diff
@@ -0,0 +1,135 @@
+.. _nrf_modem_lib_readme:
+
+Modem library integration layer
+###############################
+
+.. contents::
+   :local:
+   :depth: 2
+
+
+The Modem library integration layer handles the integration of the Modem library into |NCS|.
+The integration layer is constituted by the library wrapper and functionalities like socket offloading, OS abstraction, memory reservation by the Partition manager, and diagnostics.
+
+Library wrapper
+***************
+
+The library wrapper provides an encapsulation over the core Modem library functions such as initialization and shutdown.
+The library wrapper is implemented in :file:`nrf\\lib\\nrf_modem_lib\\nrf_modem_lib.c`.
+
+The library wrapper encapsulates the :c:func:`nrf_modem_init` and :c:func:`nrf_modem_shutdown` calls of the Modem library with :c:func:`nrf_modem_lib_init` and :c:func:`nrf_modem_lib_shutdown` calls, respectively.
+The library wrapper eases the task of initializing the Modem library by automatically passing the size and address of all the shared memory regions of the Modem library to the :c:func:`nrf_modem_init` call.
+
+The library wrapper can also initialize the Modem library during system initialization using :c:macro:`SYS_INIT`.
+The :option:`CONFIG_NRF_MODEM_LIB_SYS_INIT` Kconfig option can be used to control the initialization.
+Some libraries in |NCS|, such as the :ref:`at_cmd_readme` and the :ref:`lte_lc_readme` have similar options to initialize during system initialization and these options depend on the Modem library glue option.
+If your application performs an update of the nRF9160 modem firmware, you must disable this functionality to have full control on the initialization of the library.
+
+:ref:`partition_manager` is the component that reserves the RAM memory for the Modem library.
+For more information, see :ref:`partition_mgr_integration`.
+
+The library wrapper also coordinates the shutdown operation among different parts of the application that use the Modem library.
+This is done by the :c:func:`nrf_modem_lib_shutdown` function call, by invoking sleeping threads when the modem is being shut down.
+
+When using the Modem library in |NCS|, the library should be initialized and shutdown using the :c:func:`nrf_modem_lib_init` and :c:func:`nrf_modem_lib_shutdown` function calls, respectively.
+
+Socket offloading
+=================
+
+Zephyr Socket API offers the :ref:`socket offloading functionality <zephyr:net_socket_offloading>` to redirect or *offload* function calls (for example, ``socket()`` and ``send()``) to BSD socket APIs.
```

Comment:
I would rephrase this slightly because the BSD socket APIs are actually the APIs being offloaded here, so perhaps: ".. offload function calls to BSD socket APIs, for example, socket() and send()"

Reply:
Fixed